src-generate/Generator.hs

```diff
@@ -461,6 +462,19 @@ getInsertFunc maxCapability typeDesc = unlines ["{-# INLINE " ++ funcName ++ " #
         toCasesHelper (_, x) [] = ["| otherwise " ++ x]
         toCasesHelper (c, x) xs = ("| " ++ c ++ " " ++ x) : xs
 
+-- | Generate a function to generate a vector from an index-function
+getGenerateFunc :: TypeDesc -> String
+getGenerateFunc typeDesc = unlines
+    ["{-# INLINE[1] " ++ funcName ++ " #-}"
```

Comment:
Seems reasonable. I will try to look at it soon, but can not tell you exactly when this will be the case. Regarding the inline pragma, I would say that the aim should be that the compiler generates something of type `Int# -> Double#` doing the actual calculation and then passes this to some `generateVector# :: (Int# -> Double#) -> DoubleXN#` function, so the important steps exist in an unboxed fashion. This way most boxing/unboxing steps regarding other operations can normally be avoided / removed by the compiler.